sklearn/calibration.py

```diff
@@ -266,30 +281,32 @@ def fit(self, X, y, sample_weight=None):
 
             cv = check_cv(self.cv, y, classifier=True)
             fit_parameters = signature(base_estimator.fit).parameters
-            supports_sw = "sample_weight" in fit_parameters
-
+            base_estimator_supports_sw = "sample_weight" in fit_parameters
+            sample_weight_base_estimator = None
             if sample_weight is not None:
                 sample_weight = _check_sample_weight(sample_weight, X)
 
-                if not supports_sw:
+                if not base_estimator_supports_sw:
                     estimator_name = type(base_estimator).__name__
                     warnings.warn("Since %s does not support sample_weights, "
                                   "sample weights will only be used for the "
                                   "calibration itself." % estimator_name)
+                else:
+                    sample_weight_base_estimator = sample_weight
```

Comment:
In this specific case,  is `sample_weight_base_estimator` always equal to `sample_weight`? If this is the case, I do not think we need a `sample_weight_base_estimator` parameter in `_fit_calibrated_classifer`.